### PR TITLE
Fixes#82 Update c Code from https://github.com/panstromek/zebra

### PR DESCRIPTION
**Pull request by oers**
Newest ndk compiler lead to crashes. Bugfixes from panstromek solved the problem.s

### CODE REVIEW

**Comment by oers** [comment]
Fixes https://github.com/oers/reversatile/issues/82

File: project/src/main/jni/zebra/macros.h [hunk 1]
```diff
@@ -36,21 +36,25 @@ extern "C" {
 
 /* Define the inline directive when available */
 #if defined( __GNUC__ )&& !defined( __cplusplus )
-#define INLINE __inline__
+#define INLINE
```
Comment (panstromek) [inline project/src/main/jni/zebra/macros.h:39]: Note: I had to remove this to make c2rust transpilation work, but it should work here. But these days, it should be mostly redundant, as compilers make much better decisions about inlining anyway.

File: project/src/main/jni/zebra/macros.h [hunk 2]
```diff
@@ -36,21 +36,25 @@ extern "C" {
...
+#ifdef _X64_
+    #if __GNUC__ >= 3
+        #define	REGPARM(num)	__attribute__((regparm(num)))
+    #else
+        #if defined (_MSC_VER) || defined(__BORLANDC__)
+            #define	REGPARM(num)	__fastcall
+        #else
+            #define	REGPARM(num)
+        #endif
+    #endif
```
Comment (panstromek) [inline project/src/main/jni/zebra/macros.h:55]: Similar to `__inline__`, if you ever have any problem with these, just remove them, these are pretty outdated optimization flags. I doubt they make a difference these days.

File: project/src/main/jni/zebra/zebra.c [hunk 3]
```diff
@@ -609,8 +609,13 @@ main( int argc, char *argv[] ) {
   global_setup( use_random, hash_bits );
   init_thor_database();
 
-  if ( use_book )
-    init_learn( "book.bin", TRUE );
+  if ( use_book ) {
+      char *book_filename = getenv("BOOK_PATH");
+      if (book_filename == NULL) {
+          book_filename = "book.bin";
+      }
```
Comment (panstromek) [inline project/src/main/jni/zebra/zebra.c:616]: I changed this just for testing but it doesn't hurt here..

**Comment by panstromek** [comment]
> Newest ndk compiler lead to crashes.

yea, that makes a lot of sense. Most of these were UB fixes, and compiler is free to exploit UB for optimizations, so new optimizations can break more things.

File: project/src/main/jni/zebra/macros.h (hunk 2 above)
Comment (oers) [reply project/src/main/jni/zebra/macros.h:55]: this stuff was everywhere, maybe we can remove this in your repo

**Pull request merged**